plugin/web/extensions/sidebar/core/components/desktop/hooks/datasetHooks.ts

```diff
@@ -119,31 +142,36 @@ export default ({
       updateProject?.(project => {
         const updatedDatasets = [...project.datasets];
         const datasetIndex = updatedDatasets.findIndex(d2 => d2.dataID === dataID);
+
         if (datasetIndex >= 0) {
           updatedDatasets[datasetIndex] = dataset;
         }
+
         return {
           ...project,
           datasets: updatedDatasets,
         };
       });
 
-      handleDataRequest(dataset);
-
-      if (publish && publishToGeospatial && dataset.itemId && backendURL && backendAccessToken) {
-        fetch(`${backendURL}/publish_to_geospatialjp`, {
-          headers: {
-            authorization: `Bearer ${backendAccessToken}`,
-            "Content-Type": "application/json",
-          },
-          method: "POST",
-          body: JSON.stringify({ id: dataset.itemId }),
-        })
-          .then(r => {
-            if (!r.ok)
-              throw `failed to publish the data on gspatial.jp: status code is ${r.statusText}`;
-          })
-          .catch(console.error);
+      try {
+        await handleDataRequest(dataset);
```

Comment:
Any reason this handleDataRequest is inside the try/catch? Doesn't this mean if the handleDataRequest fails we will never do the geospatialjp fetch? That isn't good